src/diffusers/models/dualencoder_gfn.py

```diff
@@ -0,0 +1,695 @@
+# Model adapted from GeoDiff https://github.com/MinkaiXu/GeoDiff
+from typing import Callable, Dict, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+from torch.nn import Embedding, Linear, Module, ModuleList, Sequential
+
+from rdkit.Chem.rdchem import BondType as BT
+from torch_geometric.nn import MessagePassing, radius, radius_graph
```

Comment:
We need to make sure that this model is only imported if `torch_geometric` or `torch_scatter` is present. We should therefore adapt the same logic we did for the optional `transformers` import. 

Could you do the following:

1. Add a `is_torch_scatter_available` and a `is_torch_geometric_available` to https://github.com/huggingface/diffusers/blob/main/src/diffusers/utils/__init__.py (just like it has been done for other dependencies)
2. Import `DualEncoderEpsNetwork` only if both dependencies are available here: https://github.com/huggingface/diffusers/blob/89f2011cede1a899cc3f7e4b47ae1178d3a6e68c/src/diffusers/models/__init__.py#L21 (just like we do it for LDM in pipelines here: https://github.com/huggingface/diffusers/blob/89f2011cede1a899cc3f7e4b47ae1178d3a6e68c/src/diffusers/pipelines/__init__.py#L9
3. Ony import the model if the depedency is available in the main init as well: https://github.com/huggingface/diffusers/blob/89f2011cede1a899cc3f7e4b47ae1178d3a6e68c/src/diffusers/__init__.py#L26 - otherwise import the dummy class
4. Having written 3.) run `make fix-styles` to automatically create the dummy class

Reply:
So the painpoint here is that is needs specific versions too, because this code was made before breaking changes in torch_geometric. Can I do similar functions for that?